test/integration/features/profile.feature

```diff
@@ -2,104 +2,251 @@
+      When executing "minishift --profile minishift config get disk-size" succeeds
+       Then stdout should match "<nil>"
+      When executing "minishift --profile minishift config get cpus" succeeds
+       Then stdout should match "<nil>"
+      When executing "minishift --profile minishift addons list"
+       Then stdout should match "admin-user\s*: disabled\s*P\(0\)"
+       And stdout should match "anyuid\s*: disabled\s*P\(0\)"
+       And stdout should match "registry-route\s*: disabled\s*P\(0\)"
+       And stdout should match "xpaas\s*: disabled\s*P\(0\)"
 
-      """
-    When executing "minishift profile set minishift" succeeds
+  Scenario: As a user, I can start profile 'foo'
+    Given Profile "foo" has state "Does Not Exist"
+     And executing "minishift profile set foo" succeeds
+     Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift start" succeeds
       Then stdout should contain
       """
-      Profile 'minishift' set as active profile.
+      Add-on 'registry-route' created docker-registry route.
       """
-Scenario: User should not be able to create profile with blank profile name
-    When executing "minishift profile set"
-     Then exitcode should equal "1"
-     Then stderr should contain
+      And Profile "foo" should have state "Running"
+  
+  @minishift-only
+  Scenario: As user, I can list add-ons in profile 'foo' 
+    Given Profile "foo" has state "Running"
+     And executing "minishift profile set foo" succeeds
+     Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift addons list"
+     Then stdout should match "admin-user\s*: disabled\s*P\(0\)"
+      And stdout should match "anyuid\s*: disabled\s*P\(0\)"
+      And stdout should match "registry-route\s*: enabled\s*P\(0\)"
+      And stdout should match "xpaas\s*: disabled\s*P\(0\)"
+
+  Scenario: Checking that memory value was applied on profile 'foo'
+    Given Profile "foo" has state "Running"
+     And executing "minishift profile set foo" succeeds
+     Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift ssh -- less /proc/meminfo" succeeds
+      Then stdout should match "MemTotal:\s*4[0-1][0-9]{5}\s*kB"
+
+  Scenario: Checking that disk-size value was applied on profile 'foo'
+    Given Profile "foo" has state "Running"
+     And executing "minishift profile set foo" succeeds
+     Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift ssh -- sudo fdisk -l | grep Disk" succeeds
+      Then stdout should match "Disk \/dev\/sda: 3[0-2]\.?[0-9]{0,2} (GB|GiB)"
+
+  Scenario: Checking that cpus value was applied on profile 'foo'
+    Given Profile "foo" has state "Running"
+     And executing "minishift profile set foo" succeeds
+      Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
```

Comment:
`Given, And, Then, When` is not right. Should be `Given, And, And...`, `When, And, And...`, `Then, And, And...`. We might change the matching groups if the step sounds bad with `Given` statement.